File: week_1/day-3-two-pointer/variations_que/remove_element.py
nums = [3, 2, 2, 3, 4]
k = 3

def removeElement(nums, k):
    slow = 0
    for fast in range(len(nums)):
        if nums[fast] != k:
            nums[slow] = nums[fast]
            slow += 1
    return slow

length = removeElement(nums, k)
print(length)           # Output: 3
print(nums[:length])    # Output: [2, 2, 4]

File: week_1/day-3-two-pointer/variations_que/remove_element.js
const nums = [3, 2, 2, 3, 4];
const k = 3;

function removeElement(nums, k) {
    let slow = 0;
    for (let fast = 0; fast < nums.length; fast++) {
        if (nums[fast] !== k) {
            nums[slow++] = nums[fast];
        }
    }
    return slow;
}

const length = removeElement(nums, k);
console.log(length);              // 3
console.log(nums.slice(0, length)); // [2, 2, 4]
